source/main.cpp

```diff
@@ -270,6 +272,7 @@ namespace Aws
 
 int main(int argc, char *argv[])
 {
+    LockFileUtils::ProcessLock();
```

Comment:
we should lock file after MQTT connection is established since `abort()` is called if MQTT connection fails to establish.

Reply:
I am not sure if this will lead to desired behavior. Establishing a new MQTT connection with an existing one already running will cause both connections to disconnect and attempt reconnection, which is the original problem acquiring the lock is trying to prevent.